### PR TITLE
add: Enable follower fetching via API

### PR DESCRIPTION
**Pull request by harshini-rangaswamy**
## Describe your changes
Steps on how to enable follower fetching via API and Terraform
## Checklist

- [x] The first paragraph of the page is on one line.
- [x] The other lines have a line break at 90 characters.
- [x] I checked the output.
- [x] I applied the [style guide](../styleguide.md).
- [x] My links start with `/docs/`.

### CODE REVIEW

**Review by gpkc — APPROVED** [review]
LGTM!

File: docs/products/kafka/howto/enable-follower-fetching.md [hunk 1]
```diff
@@ -22,21 +28,114 @@ physical location might have different AZ names in different accounts. To ensure
...
+    kafka {
+      # Other Kafka configurations...
+
+      user_config {
+        follower_fetching = {
+          enabled = true
+        }
+      }
+     }
+    }
```
Comment (gpkc) [inline docs/products/kafka/howto/enable-follower-fetching.md:125]: I think it should be

```suggestion
   resource "aiven_kafka" "example_kafka" {
    project      = "YOUR_PROJECT_NAME"
    cloud_name   = "cloud_region"
    plan         = "business-4"
    service_name = "example-service-name"

    kafka_user_config {
      # Other Kafka configurations...

      follower_fetching = {
        enabled = true
      }
     }
    }
```

based on https://registry.terraform.io/providers/aiven/aiven/latest/docs/resources/kafk

Comment (harshini-rangaswamy) [reply docs/products/kafka/howto/enable-follower-fetching.md]: Thanks! Updated the code snippet.

File: docs/products/kafka/howto/enable-follower-fetching.md [hunk 2]
```diff
@@ -1,15 +1,21 @@
...
+
+Enabling follower fetching in Aiven for Apache Kafka® allows your consumers to fetch data from the nearest replica instead of the leader, optimizing data fetching and enhancing performance.
 
 ## Prerequisites
 
 - Aiven for Apache Kafka service version 3.6 or later.
 - [Availability zone (AZ)](#identify-availability-zone) information for your
   Aiven for Apache Kafka service.
+- [Aiven CLI](/docs/tools/cli) client.
+- [Aiven Provider for Terraform](https://registry.terraform.io/providers/aiven/aiven/latest/docs)
```
Comment (ArthurFlag) [inline docs/products/kafka/howto/enable-follower-fetching.md:17]: Could we harmonize punctuation?

File: docs/products/kafka/howto/enable-follower-fetching.md [hunk 3]
```diff
@@ -22,21 +28,111 @@ physical location might have different AZ names in different accounts. To ensure
...
+- `YOUR_PROJECT_NAME`: Name of your project.
+- `YOUR_SERVICE_NAME`: Name of your service.
+- `YOUR_BEARER_TOKEN`: API token for authentication.
 - `follower_fetching={"enabled": true}`: Enables the follower fetching feature.
 
+</TabItem>
+<TabItem value="terraform" label="Terraform">
+
+1. Add the Aiven Terraform provider to the `required_providers` block in your Terraform
+configuration:
```
Comment (ArthurFlag) [inline docs/products/kafka/howto/enable-follower-fetching.md:83]: indention issue :)

File: docs/products/kafka/howto/enable-follower-fetching.md [hunk 4]
```diff
@@ -22,21 +28,111 @@ physical location might have different AZ names in different accounts. To ensure
...
+   ```hcl
+   provider "aiven" {
+    api_token = "YOUR_API_TOKEN"
+    }
+   ```
+
+   The `api_token` is your Aiven API token for [authentication](/docs/platform/howto/create_authentication_token).
+
+1. Enable follower fetching in your Aiven for Apache Kafka service using the
+following configuration:
```
Comment (ArthurFlag) [inline docs/products/kafka/howto/enable-follower-fetching.md:107]: (indentation)

File: docs/products/kafka/howto/enable-follower-fetching.md [hunk 5]
```diff
@@ -22,21 +28,111 @@ physical location might have different AZ names in different accounts. To ensure
...
+    }
+   ```
+
+   Parameters:
+
+    - `project`: Name of your project.
+    - `cloud_name`: Cloud region where the service is hosted.
+    - `plan`: Service plan.
+    - `service_name`: Name of your Kafka service.
+    - `follower_fetching.enabled`: Set to `true` to enable the follower fetching feature.
```
Comment (ArthurFlag) [inline docs/products/kafka/howto/enable-follower-fetching.md:131]:
```suggestion
   - `project`: Name of your project.
   - `cloud_name`: Cloud region where the service is hosted.
   - `plan`: Service plan.
   - `service_name`: Name of your Kafka service.
   - `follower_fetching.enabled`: Set to `true` to enable the follower fetching feature.
```

**Review by ArthurFlag — CHANGES_REQUESTED** [review]
minor nitpicks added, should we also add the image?

**Comment by harshini-rangaswamy** [comment]
> minor nitpicks added, should we also add the image?

Addressed the indentations. 
The image was added to the concept topic - https://aiven.io/docs/products/kafka/concepts/follower-fetching

**Review by ArthurFlag — APPROVED** [review]

**Review by gpkc — APPROVED** [review]

**Pull request merged**